pyoozie/builder.py

```diff
@@ -27,56 +28,77 @@ def _coordinator_submission_xml(username, coord_xml_path, configuration=None, in
 
 class WorkflowBuilder(object):
 
-    def __init__(self, name):
-        # Initially, let's just use a static template and only one action payload and one action on error
-        self._name = tags.validate_xml_name(name)
-        self._action_name = None
-        self._action_payload = None
-        self._action_error = None
-        self._kill_message = None
-
-    def add_action(self, name, action, action_on_error, kill_on_error='${wf:lastErrorNode()} - ${wf:id()}'):
-        # Today you can't rename your action and you can only have one, but in the future you can add multiple
-        # named actions
-        if any((self._action_name, self._action_payload, self._action_error, self._kill_message)):
-            raise NotImplementedError("Can only add one action in this version")
-        else:
-            self._action_name = tags.validate_xml_id('action-' + name)
-            self._action_payload = action
-            self._action_error = action_on_error
-            self._kill_message = kill_on_error
+    def __init__(self, name, action_layout_strategy=transforms.serial_layout, parameters=None,
+                 configuration=None, credentials=None, job_tracker=None, name_node=None, job_xml_files=None,
+                 default_retry_max=None, default_retry_interval=None, default_retry_policy=None):
```

Comment:
Let's accept all of the fields that a workflow tag needs and also an `action_layout_strategy` function (e.g. do we want our nodes arranged serially? Do we want forks/joins according to some alg?).